srtp/srtp.c

```diff
@@ -4750,3 +4820,86 @@ srtp_err_status_t srtp_get_stream_roc(srtp_t session,
 
     return srtp_err_status_ok;
 }
+
+#ifndef SRTP_NO_STREAM_LIST
+
+/* in the default implementation, we have an intrusive linked list */
+struct srtp_stream_list_ctx_t_ {
+    srtp_stream_ctx_t data;
+} srtp_stream_list_ctx_t_;
+
+srtp_err_status_t srtp_stream_list_alloc(srtp_stream_list_t *list_ptr)
+{
+    srtp_stream_list_t list =
+        srtp_crypto_alloc(sizeof(srtp_stream_list_ctx_t_));
+    if (list == NULL) {
+        return srtp_err_status_alloc_fail;
+    }
+    *list_ptr = list;
```

Comment:
```suggestion
    list->prev = NULL;
    list->next = NULL;

    *list_ptr = list;
```

Reply:
Actually, what is the right "empty list" state?  On the one hand, `srtp_stream_list_dealloc` assumes `NULL`.  But if we want this to actually be a proper doubly-linked list, traversable from either end, it seems like the empty list state should have `data.prev == data.next == &data`.  This also avoids the need for `stream->next == NULL` checks.

Reply:
All that said, why do we need the double linkage here?  It seems like libsrtp only ever needs to traverse in one direction.

Reply:
srtp_crypto_alloc  will have set the struct to zero, so setting NULL here is not necessary but is maybe ok to be explicit.

The double linkage is only to facilitate fast removal, ie to get hold of the previous item. There is no intention to traverse in revers.